python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py

```diff
@@ -101,12 +132,33 @@ def _execUpdateAction(self):
                 else:
                     action.callback_Failure()
 
+            self._currently_running_command = False
+
 # Create the thread pool
 
 
 def _makeThreadPool(threadPoolSize=THREAD_POOL_SIZE, daemonic=True):
+    global ThreadPool, global_start_time, heartbeat_times
+    global_start_time = time.time()
+    if ThreadPool and len(ThreadPool) != 0:
+
+        for this_thread in ThreadPool:
+            log.error("%s running: %s" % (this_thread._thread_name, this_thread._running_cmd))
+
+        #from Ganga.Core.exceptions import GangaException
+        #raise GangaException("Cannot doubbly init the ThreadPool! ThreadPool already populated with threads")
+        log.error("Found a thread pool already in existance, wiping it and startig again!")
+        del ThreadPool[:]
+        ThreadPool = []
+
+    if heartbeat_times is not None:
+        heartbeat_times = None
+
+    heartbeat_times = defaultdict( lambda: global_start_time )
```

Comment:
Not really, I wanted to do it here to avoid disabling/re-enabling problems as the heartbeat will be stable if turned on for the first time after >3min